docs/CONTRIBUTING.md

```diff
@@ -6,10 +6,9 @@
 
 The serverless functions must be split into a separate repo to streamline the inclusion of dependencies and to be able to set up CI/CD for the back-end.
 
-Not yet decided:
+The back-end must reflect our ambitions and workflow as well as our front-end is a showcase of how we work and how we want to work. Serverless fuctions in AWS Lambda's with Node are modern and scalable and fit the microservice architecture that is appropriate for the use case of calling a Twitter feed for example.
 
-* to build a back-end a traditional Node + Express server on e.g. Heroku
-* to build a back-end in Scala
+If we would implement a full scale back-end server, rather than running Node + Express, we would choose a Scala based solution to showcase the workflow of our Scala developers. This would probably be hosted on Heroku or whatever Cloud VPS we will commonly use at that time.
 
 ### Structural organization
 
```

Comment:
I've looked up some guidelines for Atomic design. Maybe they will help with extending this document?

---

**Atoms**
Stateless components who do not rely on other (internal components);
* Button
* Heading
* Icon
* Label
* Link
* Input
* Paragraph

**Molecules**
Composition of Atoms;
* Form Field: using Atoms: Label, Input
* Search Form: using Atoms: Label, Input
* Introduction: using Atoms: Heading, Paragraph

**Organisms**
Groups of Molecules;
* Navigation: using Molecules: Buttons, Links
* Header: using Molecules: Navigation, Search Form, Toggle Button
* Footer: using Molecules: Heading, Paragraph, Links, Social Icons

**Ecosystems**
Composition of one or more Organisms.
The container component between the (Redux) store / fetching data and the Organism(s) relying on this data.

**Environments**
* Root components, like `<App />`

**Sources**
* https://github.com/pixelfusion/front-end-stack/wiki/Atomic-Design#1-atoms
* https://medium.com/@yejodido/atomic-components-managing-dynamic-react-components-using-atomic-design-part-1-5f07451f261f
* https://rangle-io.gitbooks.io/react-training/content/book/thinking_in_components/atomic_component_principles.html

Reply:
Suggestion for structuring with state:

We would allow 2 variants.

1. The internal state variant. If a component does not retrieve data but has state, use recompose.withStateHandlers. Store the initialState and stateUpdaters in the component.
Export the base component and export the composed component as default.
An example would be "mouse over" state in src/components/Atoms/EventsButton

2. The external state variant. If a component retrieves data externally, store the retrieving logic in /src/containers/*. An example is:

* src/components/Molecules/NavBar/NavBar (the stateless component)
* src/containers/NavContainer (adds state to NavBar through API call with Fetch)
* src/App.jsx (uses NavContainer)

This approach is described in more detail here: https://github.com/diegohaz/arc/wiki/Containers

In the mentioned variants, "retrieve data" can mean either calling an API with Fetch or accessing a data (Redux) store. We do not have a data store at the moment.

**NOTE** src/modules should be renamed to src/containers